applications/MappingApplication/custom_utilities/mapper_utilities.h

```diff
@@ -112,189 +109,337 @@ namespace Kratos
-          noalias(vector_points) = point_to_project.Coordinates() - point_in_plane.Coordinates();
+      static double ComputeSearchRadius(ModelPart& rModelPart1, ModelPart& rModelPart2, const int EchoLevel) {
+          double search_radius = std::max(ComputeSearchRadius(rModelPart1, EchoLevel),
+                                          ComputeSearchRadius(rModelPart2, EchoLevel));
+          return search_radius;
+      }
 
-          CalculateLineNormal(p_condition, normal);
+      static double ComputeSearchRadius(ModelPart& rModelPart, const int EchoLevel) {
+          double search_safety_factor = 1.2;
+          double max_element_size = 0.0;
 
-          distance = inner_prod(vector_points, normal);
+          int num_conditions_global = ComputeNumberOfConditions(rModelPart);
 
-          point_projected.Coordinates() = point_to_project.Coordinates() - normal * distance;
+          if (num_conditions_global > 0) {
+              max_element_size = ComputeMaxEdgeLengthLocal(rModelPart.GetCommunicator().LocalMesh().Conditions());
 
-          distance = fabs(distance);
+          } else {
+              if (EchoLevel > 1 && rModelPart.GetCommunicator().MyPID() == 0)
+              std::cout << "MAPPER WARNING, no conditions/elements for search radius "
+                        << "computations in ModelPart \"" << rModelPart.Name() << "\" found, "
+                        << "using nodes (less efficient)" << std::endl;
+              max_element_size = ComputeMaxEdgeLengthLocal(rModelPart.GetCommunicator().LocalMesh().Nodes());
+          }
 
-          array_1d<double, 3> point_projected_local_coords;
-          point_projected_local_coords = p_condition->GetGeometry().PointLocalCoordinates(point_projected_local_coords,point_projected);
+          rModelPart.GetCommunicator().MaxAll(max_element_size); // Compute the maximum among the partitions
+          return max_element_size * search_safety_factor;
+      }
 
-          local_coords[0] = point_projected_local_coords[0];
-          local_coords[1] = 0.0;
+      static double ComputeConservativeFactor(const double NumNodesOrigin,
+                                              const double NumNodesDestination) {
+          // NumNodes* are casted to doubles in order to use the double devision
+          // if this function would take ints, then the return value would also be an int!
+          return NumNodesOrigin / NumNodesDestination;
+      }
 
-          std::cout << "Point to project: [" << global_coords[0] << " " << global_coords[1] << " " << global_coords[2] << "], local points: [" <<
-          p_condition->GetGeometry()[0].X() << " " << p_condition->GetGeometry()[0].Y() << " " << p_condition->GetGeometry()[0].Z() << "] ["<<
-          p_condition->GetGeometry()[1].X() << " " << p_condition->GetGeometry()[1].Y() << " " << p_condition->GetGeometry()[1].Z() << "] local coord: " <<
-          local_coords[0] << std::endl;
+      static bool ProjectPointToLine(Condition* pCondition,
```

Comment:
For this function (and the following ones) I don't see much the point of passing the `Condition` object, since you only use its underlying geometry. If you pass the geometry, you will have a function thta works for Conditions AND Elements.